src/rust/src/conversion/mod.rs

```diff
@@ -33,6 +33,25 @@ impl<T> From<T> for Wrap<T> {
     }
 }
 
+impl TryFrom<Sexp> for Wrap<AnyValue<'_>> {
+    type Error = String;
+    fn try_from(obj: Sexp) -> Result<Self, String> {
+        let typed = obj.into_typed();
+        let out = match typed {
+            TypedSexp::Integer(x) => AnyValue::Int64(*(x.to_vec().first().unwrap()) as i64),
+            TypedSexp::Real(x) => AnyValue::Float64(*(x.to_vec().first().unwrap())),
+            TypedSexp::Logical(x) => AnyValue::Boolean(*(x.to_vec().first().unwrap())),
+            TypedSexp::String(x) => {
+                let val = x.to_vec();
+                AnyValue::StringOwned((*val.first().unwrap()).into())
+            }
+            TypedSexp::Null(_) => AnyValue::Null,
+            _ => return Err("Cannot cast to AnyValue".to_string()),
+        };
+        Ok(Wrap(out))
+    }
+}
```

Comment:
What is this?
Do not add conversions that do not involve `as_polars_series()`. (i.e., this should have to be removed)

Reply:
This is the way it is in py-polars:

https://github.com/pola-rs/polars/blob/91ad299b17c48422131c584f63acedc758581100/crates/polars-python/src/lazyframe/general.rs#L992C27-L992C31

There is no other places in our codebase where an input is `Wrap<AnyValue<'_>>` so far and it's very rare in the py-polars coddebase:

https://github.com/search?q=repo%3Apola-rs%2Fpolars+Wrap%3CAnyValue%3C%27_%3E%3E&type=code

Reply:
Check carefully: does Python Polars do type mapping in more than one place?
The way the type mapping is done here is fundamentally different from that in polars-python. So this must be removed.

Reply:
To add a note that may help your understanding, in Python Polars, Python objects are converted to Polars literals via AnyValue. But that is not the case here.

Reply:
I don't understand what you mean. In `join_asof()`'s options, `tolerance` must be of type `AnyValue` so we will have to convert the input to an `AnyValue` in Rust anyway, no?
```rust
pub struct AsOfOptions {
    pub strategy: AsofStrategy,
    /// A tolerance in the same unit as the asof column
    pub tolerance: Option<AnyValue<'static>>,
    ...
}
```
Usually I can figure this out by looking at functions that you already implemented in neo-r-polars, but as I said this is the only place where `AnyValue` is used as input. I don't see how to change this, so can you point me more precisely to the change to make?

Reply:
Great, thank you 👍

Reply:
However, there is a problem, it seems that if I pass something like `list(NULL)`, there is a panic.
This is not seen when Python Polars installed from PyPI. I have not had time to dig into the cause.

Reply:
https://github.com/pola-rs/polars/issues/20321

Reply:
Thanks for taking a look at this. I did my tests on Python with DataFrame, so perhaps this hints that turning off the collect method optimization will not result in panic.